client/src/components/NotifsDrawer.js

```diff
@@ -0,0 +1,127 @@
+import React, { useState, useEffect } from "react";
+import {
+  Drawer,
+  List,
+  ListItem,
+  ListItemText,
+  Typography,
+  ListItemAvatar,
+  ListItemIcon,
+} from "@material-ui/core";
+import ChatIcon from "@material-ui/icons/Chat";
+import MonetizationOnIcon from "@material-ui/icons/MonetizationOn";
+import { Link } from "react-router-dom";
+import notificationsAPI from "../notificationsAPI";
+
+const NotifsDrawer = ({
+  notifsOpen,
+  setNotifsOpen,
+  classes,
+  setUnreadCount,
+}) => {
+  const [notifs, setNotifs] = useState([]);
+
+  useEffect(() => {
+    setNotifs(notificationsAPI());
+  }, []);
+
+  useEffect(() => {
+    setUnreadCount(notifs.length);
+  }, [notifs, setUnreadCount]);
+
+  const notificationClickHandler = (notif) => {
+    notifs.splice(
+      notifs.findIndex((n) => n._id === notif._id),
+      1
+    );
+    setNotifsOpen(false);
+  };
+
+  return (
+    <Drawer
+      variant="temporary"
+      anchor="right"
+      open={notifsOpen}
+      onClose={() => setNotifsOpen(false)}
+    >
+      <div className={classes.drawerDiv} style={{ width: 350 }}>
+        <List component="nav" aria-label="navigation">
+          <ListItem divider style={{ height: 65, textAlign: "center" }}>
+            <ListItemText primary="NOTIFICATIONS" />
+          </ListItem>
+          {notifs.length === 0 && (
+            <ListItem style={{ textAlign: "center", marginTop: "20px" }}>
+              <ListItemText secondary="You have no notifications." />
+            </ListItem>
+          )}
+          {notifs.map((notif) => {
+            if (notif.type === "message") {
```

Comment:
Could this if statement be wrapped around the ListItem's children? It looks like both messages and other types of notifs use the same ListItem props

Reply:
Yes I could do that